### ISSUE TITLE
The removal of the PowerShell execution and problems updating the assembly.cs file using Node

### ISSUE THREAD

**Issue by calmath** [colindembovsky/cols-agent-tasks#5]
Hi,

As of the latest update, all of our build are now failing both hosted and on our own build server. What I have needed to do is to put the powershell back in place though with the two additional advanced parameters. The reason is that the execution of the JS falls over trying to update the physical files, I think due to some level of permissions as the files are readonly having just got the latest from source control. It seems Powershell though is able to write to the files and hence update the version numbers correctly.

This has resolved our build though we now need to ensure we execute all builds on our own build server as the hosted server with Microsoft is no longer an option. Hence if the PowerShell can be re-introduced otherwise the permission issue needs to be resolved in the Node execution. The specific error is _(note I have removed the actual full working folder myself to '...' which is correct at execution time)_:

```
2016-03-03T11:09:31.8058929Z shell.js: internal error
2016-03-03T11:09:31.8215181Z Error: EPERM, operation not permitted 'C:\Agents\Work_1\...\Properties\AssemblyInfo.cs'
2016-03-03T11:09:31.8215181Z     at Error (native)
2016-03-03T11:09:31.8215181Z     at Object.fs.openSync (fs.js:500:18)
2016-03-03T11:09:31.8215181Z     at Object.fs.writeFileSync (fs.js:1099:15)
2016-03-03T11:09:31.8215181Z     at Object._sed (C:\Agents\tasks\VersionAssemblies\0.1.3\node_modules\shelljs\src\sed.js:39:8)
2016-03-03T11:09:31.8215181Z     at Object.sed (C:\Agents\tasks\VersionAssemblies\0.1.3\node_modules\shelljs\src\common.js:186:23)
2016-03-03T11:09:31.8215181Z     at Object.<anonymous> (C:\Agents\tasks\VersionAssemblies\0.1.3\versionAssemblies.js:44:16)
2016-03-03T11:09:31.8215181Z     at Module._compile (module.js:460:26)
2016-03-03T11:09:31.8215181Z     at Object.Module._extensions..js (module.js:478:10)
2016-03-03T11:09:31.8215181Z     at Module.load (module.js:355:32)
2016-03-03T11:09:31.8215181Z     at Function.Module._load (module.js:310:12)
```

Note it took a little time as the task actually initially hadn't updated as the copy we had before the recent update was already in a version 0.1.3 folder so there may also be a versioning issue to the task though that's not my main concern at the moment. hence the task fell over regarding a missing parameter before even getting the task started. Resolved by deleting the task folder from our build server at which point the latest version was downloaded and this more critical issue was found...

Thanks
Mark

**Comment by hangar18rip** [comment]
Hi,
Which SC system do you use ? If it is TFVC, try again removing the ReadOnly attribute on the file before doing the update ;)

**Comment by calmath** [comment]
Just noticed a change to modify the file that has been made to the versioning task now our hosted agent builds work and I will also check that our build server gets the updated task.

Thanks Colin

Just in reply to the above comment just altering the ReadOnly attribute is what is required but this needs to be handled by the task itself as the build is simply performing a get latest and then running the version task so this cannot be manually intercepted all source code is by default readonly in TFVC.

Mark

**Comment by colindembovsky** [comment]
Hi @calmath 
I'm sorry that you experienced issues - I don't typically use this task with TFVC so I hadn't tested this scenario. I've added a `chmod` to change the file to writable.

I've dropped the PowerShell implementation because I don't want to have to maintain 2 versions of the task. `Node` allows the javascript task to run on Windows and other platforms, so that's why I've gone that route.

I didn't increment the version number for the Task - I did for the whole extension. I'm slowly improving how I release updates to this extension - hopefully I'll get better at it!
